### PR TITLE
fix(env): seed the wall-reshuffle chain from the init key

### LINKED ISSUE #71: Every wall after the first hand is a global constant: `_init` never seeds `round_state.rng_key`
### Summary

In both `red_mahjong` and `no_red_mahjong`, `_init` splits a `subkey` off the
init rng but never stores it — `_make_state` is called without `rng_key=`, so
`round_state.rng_key` keeps the `State` dataclass default `jax.random.PRNGKey(0)`.
Since `Env.step()` ignores its `key` argument (`del key`) and every reshuffle
derives from `round_state.rng_key` alone (`_prepare_next_round_assets`), **every
wall after the first hand is drawn from one global chain shared by all seeds,
envs, and runs**. The second hand's deck, all four deals, and the dora
indicator are literal constants of the library.

(The no_red `_init` docstring even lists "rng key" among the variables it sets —
the split was clearly intended to be stored.)

### Repro

```python
import jax, jax.numpy as jnp, numpy as np, mahjax

env = mahjax.make("red_mahjong", round_mode="half", next_round_style="dummy_share")

def second_wall(seed):
    state = env.init(jax.random.PRNGKey(seed))
    step, rng, prev = jax.jit(env.step), jax.random.PRNGKey(seed + 1), False
    while True:
        rng, k1, k2 = jax.random.split(rng, 3)
        act = jax.random.categorical(k1, jnp.where(state.legal_action_mask, 0.0, -1e9))
        state = step(state, act.astype(jnp.int32), k2)
        term = bool(state.round_state.terminated_round)
        if prev and not term:
            return np.asarray(state.round_state.deck)
        prev = term

print(np.array_equal(second_wall(0), second_wall(424242)))  # True (!)
```

### Why this matters for RL

From hand 2 on, the full wall — including the other three players' deals — is
identifiable from public observations (`round`, `honba`, and the dora
indicator, which is a constant per hand index). A self-play policy can
therefore memorize walls, which acts as an implicit full-information oracle:
it inflates in-simulator strength and evaluation numbers, and none of it
transfers to opponents/evaluations with properly randomized walls. It also
collapses the effective wall diversity of any multi-env rollout (all envs
share one wall sequence after their first hand).

### Fix

Pass the already-split (currently dead) `subkey` into `_make_state` as
`rng_key=` in both envs. The first hand's deal and dealer are bitwise
unchanged (they only ever used the other half of the split), so same-seed
reproducibility is preserved. Draft PR incoming.

### PR DESCRIPTION
**Pull request by Jannchie**
Fixes #71.

`_init` split a `subkey` off the init rng but never stored it, so
`round_state.rng_key` stayed at the `State`-default `PRNGKey(0)`. `step()`
ignores its `key` argument, and every reshuffle derives from
`round_state.rng_key` alone — so every wall after the first hand was one
global sequence shared across seeds, envs, and runs (observed:
bitwise-identical second-hand decks for seeds 0 and 424242).

Change: pass the previously dead `subkey` into `_make_state` as `rng_key=`
(both `red_mahjong` and `no_red_mahjong`), plus regression tests:

- `rng_key` varies with the init key, is not the default constant, and stays
  deterministic for a fixed key;
- two different seeds produce different second-hand walls.

The first hand's deal and dealer are bitwise unchanged (they only ever used
the other half of the split), so same-seed episode reproducibility is
preserved — a fixed init key still reproduces the same full game.

https://claude.ai/code/session_01AJkEXmv6yurU9zkjPBzagw

### CODE REVIEW

**Comment by nissymori** [comment]
Thanks for this — the diagnosis is spot on and the repro is exactly what I needed. I confirmed the bug, and it's worse than the issue states: it's not just the second wall and not just `dummy_share`. On your base, walls 2 **and** 3 are bitwise identical across seeds 0 / 424242, in **both** envs and **both** `next_round_style` values, and `round_state.rng_key` is the constant `PRNGKey(0)` for every seed.

I also verified the fix does what it claims: all walls distinct across all four env × style combinations, and hashing every leaf of `env.init(key)` against your base shows only `rng_key` changes — first-hand deals bitwise unchanged, exactly as you wrote.

**That said, I'm going to close this PR and fix it differently.** Sorry to do that to a correct patch — but it pushed me to look at the design, and the design is where the actual problem is.

The defect isn't that `rng_key` is unseeded; it's that `round_state.rng_key` exists. `core.Env.step` already declares and documents `key: Optional[Array]`, and both envs throw it away (`del key` at `red_mahjong/env.py:309` and `:358`, `no_red_mahjong/env.py:255`). Randomness is meant to enter through `step`, [pgx-style](https://github.com/sotetsuk/pgx/blob/main/pgx/core.py); the stored key was a workaround for that never having been wired up. Seeding it correctly makes the workaround work, which is what I'd rather not cement — the state would keep carrying an RNG chain that the public API says lives in `step`.

The plan instead:

- drop `rng_key` from `RoundState` (`red_mahjong/state.py:90`, `no_red_mahjong/state.py:110`) and from `_ROUND_FIELDS` (`red_mahjong/env.py:90`, `no_red_mahjong/env.py:64`)
- thread `step`'s `key` into the round transition rather than splitting a stored key — red `env.py`
- drop the `rng_key` row from `docs/api.md:103`

#71 stays open until that lands — it's your find and the credit is yours. Thanks again for chasing it down with a concrete repro; that's what made it clear the design had drifted, not just the seeding.
